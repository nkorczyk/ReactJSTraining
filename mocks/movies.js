const movies = {
  data: [
    {
      id: 337167,
      title: 'Fifty Shades Freed',
      tagline: "Don't miss the climax",
      vote_average: 6.1,
      vote_count: 1195,
      release_date: '2018-02-07',
      poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
      overview: 'Believing they have left behind shadowy figures from their past.',
      budget: 55000000,
      revenue: 136906000,
      genres: [
        'Drama',
        'Romance',
      ],
      runtime: 106,
    },
    {
      id: 269149,
      title: 'Zootopia',
      tagline: 'Welcome to the urban jungle.',
      vote_average: 7.7,
      vote_count: 6795,
      release_date: '2016-02-11',
      poster_path: 'https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg',
      overview: "Determined to prove herself, Officer Judy Hopps, the first bunny on Zootopia's police force.",
      budget: 150000000,
      revenue: 1023784195,
      genres: [
        'Animation',
        'Adventure',
        'Family',
        'Comedy',
      ],
      runtime: 108,
    },
    {
      id: 181808,
      title: 'Star Wars: The Last Jedi',
      tagline: 'The Saga Continues',
      vote_average: 7.1,
      vote_count: 4732,
      release_date: '2017-12-13',
      poster_path: 'https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg',
      overview: 'Rey develops her newly discovered abilities with the guidance of Luke Skywalker.',
      budget: 200000000,
      revenue: 1325937250,
      genres: [
        'Fantasy',
        'Adventure',
        'Science Fiction',
      ],
      runtime: 152,
    },
    {
      id: 284054,
      title: 'Black Panther',
      tagline: 'Long live the king',
      vote_average: 7.3,
      vote_count: 3788,
      release_date: '2018-02-13',
      poster_path: 'https://image.tmdb.org/t/p/w500/uxzzxijgPIY7slzFvMotPv8wjKA.jpg',
      overview: "King T'Challa returns home from America to the reclusive, technologically advanced African.",
      budget: 200000000,
      revenue: 1245257672,
      genres: [
        'Action',
        'Adventure',
        'Fantasy',
        'Science Fiction',
      ],
      runtime: 134,
    },
    {
      id: 399055,
      title: 'The Shape of Water',
      tagline: 'A Fairy Tale for Troubled Times',
      vote_average: 7.3,
      vote_count: 3200,
      release_date: '2017-12-01',
      poster_path: 'https://image.tmdb.org/t/p/w500/k4FwHlMhuRR5BISY2Gm2QZHlH5Q.jpg',
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962.',
      budget: 19500000,
      revenue: 185545281,
      genres: [
        'Drama',
        'Fantasy',
        'Romance',
      ],
      runtime: 123,
    },
    {
      id: 333339,
      title: 'Ready Player One',
      tagline: 'A better reality awaits.',
      vote_average: 8.1,
      vote_count: 617,
      release_date: '2018-03-28',
      poster_path: 'https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg',
      overview: 'When the creator of a popular video game system dies.',
      budget: 175000000,
      revenue: 0,
      genres: [
        'Adventure',
        'Science Fiction',
        'Action',
      ],
      runtime: 140,
    },
    {
      id: 338970,
      title: 'Tomb Raider',
      tagline: 'Her legend begins',
      vote_average: 6.2,
      vote_count: 817,
      release_date: '2018-03-08',
      poster_path: 'https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg',
      overview: 'Lara Croft, the fiercely independent daughter of a missing adventurer.',
      budget: 94000000,
      revenue: 126025000,
      genres: [
        'Action',
        'Adventure',
      ],
      runtime: 118,
    },
    {
      id: 284053,
      title: 'Thor: Ragnarok',
      tagline: 'No Hammer. No Problem.',
      vote_average: 7.4,
      vote_count: 5349,
      release_date: '2017-10-25',
      poster_path: 'https://image.tmdb.org/t/p/w500/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg',
      overview: 'Thor is imprisoned on the other side of the universe.',
      budget: 180000000,
      revenue: 854229371,
      genres: [
        'Action',
        'Adventure',
        'Fantasy',
      ],
      runtime: 130,
    },
    {
      id: 321612,
      title: 'Beauty and the Beast',
      tagline: 'Be our guest.',
      vote_average: 6.8,
      vote_count: 7861,
      release_date: '2017-03-16',
      poster_path: 'https://image.tmdb.org/t/p/w500/tWqifoYuwLETmmasnGHO7xBjEtt.jpg',
      overview: "A live-action adaptation of Disney's version of the classic tale of a cursed prince.",
      budget: 160000000,
      revenue: 1263521126,
      genres: [
        'Family',
        'Fantasy',
        'Romance',
      ],
      runtime: 129,
    },
  ],
  total: 3000,
  offset: 0,
  limit: 10,
};

export default movies;
