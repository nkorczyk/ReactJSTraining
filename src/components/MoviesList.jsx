// @flow
import React from 'react';
import Item from './Item';
import styled from 'styled-components';

type Props = {
  movies: []
}

const MoviesListContainer = styled.div`
  display: grid;
  grid-template-columns: 256px 256px 256px;
  grid-gap: 10px;
  background: #fff;
  margin: 0 auto;
  padding: 5px 5px;
  max-width: 800px;
`;

const MoviesList = ({ movies }: Props) => (
  <MoviesListContainer>
    {movies.map(movie => (<Item movie={movie} key={movie.id} />))}
  </MoviesListContainer>
);

export default MoviesList;
