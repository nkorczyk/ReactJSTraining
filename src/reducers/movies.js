import initialState from './initialState';
import ACTION_TYPES from '../actions/types';

function sortby(state = initialState.movies, action) {
  switch (action.type) {
    case ACTION_TYPES.LOAD_MOVIES:
      state = {
        ...state,
        status: ACTION_TYPES.LOAD_MOVIES
      };
      return state;
    case ACTION_TYPES.LOAD_MOVIES_SUCCESS:
      state = {
        ...state,
        data: action.movies,
        status: ACTION_TYPES.LOAD_MOVIES_SUCCESS
      };
      return state;
    case ACTION_TYPES.LOAD_MOVIES_ERROR:
      state = {
        data: [],
        status: ACTION_TYPES.LOAD_MOVIES_ERROR
      };
      return state;
    default:
      return state;
  }
}

export default sortby;
