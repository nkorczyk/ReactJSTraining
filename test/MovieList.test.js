import React from 'react';
import { shallow } from 'enzyme';
import MoviesList from '../src/components/MoviesList';
import movies from '../mocks/movies';

describe('MoviesList', () => {
  it('should rendered correctly without any movie', () => {
    const mockedMovies = [];
    const component = shallow(<MoviesList movies={mockedMovies} />);
    expect(component).toMatchSnapshot();
  });

  it('should rendered correctly with 5 movies', () => {
    const mockedMovies = movies.data.splice(0, 5);
    const component = shallow(<MoviesList movies={mockedMovies} />);
    expect(component).toMatchSnapshot();
  });
});
