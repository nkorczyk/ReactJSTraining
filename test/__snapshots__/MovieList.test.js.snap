// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList should rendered correctly with 5 movies 1`] = `
<div
  className="movies-list"
>
  <Item
    key="337167"
    movie={
      Object {
        "budget": 55000000,
        "genres": Array [
          "Drama",
          "Romance",
        ],
        "id": 337167,
        "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
        "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
        "release_date": "2018-02-07",
        "revenue": 136906000,
        "runtime": 106,
        "tagline": "Don't miss the climax",
        "title": "Fifty Shades Freed",
        "vote_average": 6.1,
        "vote_count": 1195,
      }
    }
  />
  <Item
    key="269149"
    movie={
      Object {
        "budget": 150000000,
        "genres": Array [
          "Animation",
          "Adventure",
          "Family",
          "Comedy",
        ],
        "id": 269149,
        "overview": "Determined to prove herself, Officer Judy Hopps, the first bunny on Zootopia's police force, jumps at the chance to crack her first case - even if it means partnering with scam-artist fox Nick Wilde to solve the mystery.",
        "poster_path": "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
        "release_date": "2016-02-11",
        "revenue": 1023784195,
        "runtime": 108,
        "tagline": "Welcome to the urban jungle.",
        "title": "Zootopia",
        "vote_average": 7.7,
        "vote_count": 6795,
      }
    }
  />
  <Item
    key="181808"
    movie={
      Object {
        "budget": 200000000,
        "genres": Array [
          "Fantasy",
          "Adventure",
          "Science Fiction",
        ],
        "id": 181808,
        "overview": "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
        "poster_path": "https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
        "release_date": "2017-12-13",
        "revenue": 1325937250,
        "runtime": 152,
        "tagline": "The Saga Continues",
        "title": "Star Wars: The Last Jedi",
        "vote_average": 7.1,
        "vote_count": 4732,
      }
    }
  />
  <Item
    key="284054"
    movie={
      Object {
        "budget": 200000000,
        "genres": Array [
          "Action",
          "Adventure",
          "Fantasy",
          "Science Fiction",
        ],
        "id": 284054,
        "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan \\"special forces\\"), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
        "poster_path": "https://image.tmdb.org/t/p/w500/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
        "release_date": "2018-02-13",
        "revenue": 1245257672,
        "runtime": 134,
        "tagline": "Long live the king",
        "title": "Black Panther",
        "vote_average": 7.3,
        "vote_count": 3788,
      }
    }
  />
  <Item
    key="399055"
    movie={
      Object {
        "budget": 19500000,
        "genres": Array [
          "Drama",
          "Fantasy",
          "Romance",
        ],
        "id": 399055,
        "overview": "An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.",
        "poster_path": "https://image.tmdb.org/t/p/w500/k4FwHlMhuRR5BISY2Gm2QZHlH5Q.jpg",
        "release_date": "2017-12-01",
        "revenue": 185545281,
        "runtime": 123,
        "tagline": "A Fairy Tale for Troubled Times",
        "title": "The Shape of Water",
        "vote_average": 7.3,
        "vote_count": 3200,
      }
    }
  />
</div>
`;

exports[`MoviesList should rendered correctly without any movie 1`] = `
<div
  className="movies-list"
/>
`;
